core/src/main/java/org/testcontainers/containers/GenericContainer.java

```diff
@@ -425,8 +434,25 @@ private void tryStart(Instant startedAt) {
             // For all registered output consumers, start following as close to container startup as possible
             this.logConsumers.forEach(this::followOutput);
 
+            // Wait until inspect container returns the mapped ports
+            Callable<Integer> inspectMappedExposedPortsCount = () -> {
+                containerInfo = dockerClient.inspectContainerCmd(containerId).exec();
+                return (int) containerInfo
+                    .getNetworkSettings()
+                    .getPorts()
+                    .getBindings()
+                    .values()
+                    .stream()
+                        .filter(Objects::nonNull)
+                        .count();
+            };
+
+            await()
+                .atMost(5, SECONDS)
+                .pollInterval(fibonacci(MILLISECONDS))
```

Comment:
let's also use `pollInSameThread()`, to avoid an overhead of thread creation on first poll